python/ray/serve/tests/test_deployment_state.py

```diff
@@ -212,10 +221,18 @@ def mock_deployment_state() -> Tuple[DeploymentState, Mock, Mock]:
         def mock_save_checkpoint_fn(*args, **kwargs):
             pass
 
-        deployment_state = DeploymentState(
-            "name", "name", True, mock_long_poll, mock_save_checkpoint_fn
-        )
-        yield deployment_state, timer
+        if request.param:
+            deployment_state = DriverDeploymentState(
```

Comment:
is there other special logic we should test DriverDeploymentState? like new nodes, remove nodes, etc....

Reply:
new nodes & remove nodes, it is been tested under the test_grpc.py

Reply:
that's not quite enough. can we unit test these bits? so that in the future, changing anything related to deployment state only need to re-run this test locally to verify it's working? generally we want tests to be self contained and verifying a behavior surgically.

Reply:
that make sense, added test.